src/trace/span-inferrer.ts

```diff
@@ -80,6 +96,8 @@ export class SpanInferrer {
       method = event.requestContext.http.method;
     }
     const resourceName = [method || domain, resourcePath].join(" ");
+    const serviceName = this.serviceMapping.api_gateway || domain;
```

Comment:
Your implementation here is correct, but I'm curious about the overlap with dd-trace-js using the same ENV var for service name remapping.

If I have a function triggered by API Gateway, but my lambda function also makes a request to any API Gateway endpoint, seems (or sqs, sns, etc), we need to make sure we don't accidentally overwrite the downstream spans.

Potentially we can get around this with a prefix like `lambda.apiGateway`

Reply:
We remap based on plugin names, meaning
` DD_SERVICE_MAPPING: "aws-sdk:my-new-service-name"`

on a lambda such as:

```
'use strict';
const AWS = require('aws-sdk');
const sqs = new AWS.SQS({region: 'sa-east-1'}); // Instantiate the SQS service client

module.exports.handler = async (event, context) => {
  try {
    // Send a message to an SQS queue
    const sqsParams = {
      QueueUrl: 'foobar', // Replace with your Queue URL
      MessageBody: 'Information about the current operation'
    };
    const sqsResponse = await sqs.sendMessage(sqsParams).promise();

    return {
      statusCode: 200,
      body: JSON.stringify({
        apiGatewayResponse: apiGatewayResponse.items,
        sqsResponse
      }),
    };
  } catch (err) {
    return {
      statusCode: 500,
      body: JSON.stringify(err),
    };
  }
};

```

has this effect

<img width="548" alt="Screen Shot 2023-05-11 at 5 12 51 PM" src="https://github.com/DataDog/datadog-lambda-js/assets/49878080/30856b55-6c8b-41b3-8678-479e52adde2e">

So the prefix shouldn't be necessary

Reply:
Meaning our implementation of renaming inferred spans service names should not rename any service names downstream.